src/twisted/conch/test/test_insults.py

```diff
@@ -201,19 +204,28 @@ def testByte%(groupName)s(self):
 
     self.verifyResults(transport, proto, parser)
 """
+
+
+
 class ByteGroupingsMixin(MockMixin):
-    protocolFactory = None
+    protocolFactory = None  # type: Optional[Type[Protocol]]
```

Comment:
In Python 3.6, you could annotate the variable without assigning to it:
```py
class ByteGroupingsMixin(MockMixin):
    protocolFactory: Type[Protocol]
```
I don't know if there is an alternative syntax for older Python versions, but if there isn't, this could be a motivation to move to 3.6 now.

Reply:
@adiroiban I wasn't sure whether to change these usages of `None` in base classes or not.  For these particular cases, I took the conservative approach and just used `Optional` for the annotation.

@mthuurne I brought up the topic of dropping Python 3.5 support on the mailing list:
https://twistedmatrix.com/pipermail/twisted-python/2020-May/065066.html
No decision on that has been made yet, but @glyph and @hawkowl seem OK with the idea.  If Python 3.6 can become the minimal version of Python for Twisted, then Twisted can definitely use PEP-0526 variable annotations.

Reply:
This conversation is a strong endorsement for why this change shouldn't be mixed in with a bunch of totally dissimilar changes, by the way.

Reply:
@mthuurne Good point. You can't do that in 3.5 with a type comment?

One problem is that not assigning to `None` is arguably a compatibility issue? Some code might check for `None` to see if it's been set…

An option that doesn't change the code is to set the type to `Type[Protocol]` and also use a `# type: ignore` comment to suppress the assignment error.  That correctly acknowledges the status quo and leave the question of whether to change the behavior to a separate PR.

I think that when add typing to a large codebase like Twisted, that may be the better strategy in a lot of cases; we can seek and destroy all the `# type: ignore` comments *after* documenting the current code behavior with accurate type hints.

Reply:
> One problem is that not assigning to None is arguably a compatibility issue? Some code might check for None to see if it's been set…

Thanks for your input.

In this particular usage, this is just testing code.
But yes, for compatibility reasons, I think that for the general code None should be accepted in many places.

I was thinking that we add strict typing in Twisted so that when an user of the library will run a type checker on the code the errors will help write a better code.

This is all very subjecting:  for a better code, an implementation should have dedicated `isStarted` attributes and don't use the value of protocolFactory to decide if something is started or not.